source/utilities.cc

```diff
@@ -713,31 +713,90 @@ namespace aspect
 
           i++;
 
-          // TODO: add checks for coordinate ordering in data files
         }
 
       AssertThrow(i == (components + dim) * data_table.n_elements(),
                   ExcMessage (std::string("Number of read in points does not match number of expected points. File corrupted?")));
 
+      // In case the data is specified on a grid that is equidistant
+      // in each coordinate direction, we only need to store
+      // (besides the data) the number of intervals in each direction and
+      // the begin- and endpoints of the coordinates.
+      // In case the grid is not equidistant, we need to keep
+      // all the coordinates in each direction, which is more costly.
+      // Here we fill the data structures needed for both cases,
+      // and check whether the coordinates are equidistant or not.
+      // We also check the requirement that the coordinates are
+      // strictly ascending.
+
+      // The number of intervals in each direction
       std_cxx11::array<unsigned int,dim> table_intervals;
 
+      // Whether or not the grid is equidistant
+      bool equidistant_grid = true;
+
       for (unsigned int i = 0; i < dim; i++)
         {
           table_intervals[i] = table_points[i] - 1;
 
           TableIndices<dim> idx;
-          grid_extent[i].first = data_tables[i](idx);
-          idx[i] = table_points[i] - 1;
-          grid_extent[i].second = data_tables[i](idx);
+          double temp_coord = data_tables[i](idx);
+          double new_temp_coord = 0;
+
+          // The minimum coordinates
+          grid_extent[i].first = temp_coord;
+
+          // The first coordinate value
+          coordinate_values[i].push_back(temp_coord);
+
+          // The grid spacing
+          double first_grid_spacing = 0, grid_spacing = 0;
```

Comment:
We usually split these declarations into two lines, and I do not think you need to initialize these variables (this is my mandatory comment, which proofs that I have looked through the code :smile:).